src/commands/search.ts

```diff
@@ -5,51 +5,35 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import {prompt, registerPrompt} from 'inquirer'
 import {Command, toConfiguredId, toStandardizedId} from '@oclif/core'
-import {AutocompleteSearch, Choices} from '../util/autocomplete'
+import autocomplete from 'inquirer-autocomplete-standalone'
 
 export default class Search extends Command {
-  public static summary = 'Search for a command.'
   public static description = 'Once you select a command, hit enter and it will show the help for that command.'
+  public static summary = 'Search for a command.'
 
   public async run(): Promise<unknown> {
-    const commands = (this.config.commands as Array<Command.Loadable & { readableId: string }>).filter(c => !c.hidden)
-
-    for (const command of commands) {
-      command.readableId = toConfiguredId(command.id, this.config)
-    }
-
-    registerPrompt('autocomplete-search', AutocompleteSearch)
-
-    const {command} = await prompt<{ command: string }>([
-      {
-        type: 'autocomplete-search',
-        name: 'command',
-        message: 'Search for a command',
-        emptyText: 'Nothing found!',
-        pageSize: 10,
-        validate: (val: string): string | boolean => {
-          return val ? true : 'Type something!'
-        },
-        searchOpts: {
-          keys: ['readableId'],
-          shouldSort: true,
-          threshold: 0.25,
-          ignoreLocation: true,
-        },
-        displayKey: 'readableId',
-        options: commands,
-        footer: (currentChoices: Choices, index: number): string | undefined => {
-          const cmd = commands.find(c => c.readableId === currentChoices.getChoice(index).value)
-          if (cmd && cmd.summary) return cmd.summary
-        },
+    const commands = this.config.commands
+      .filter((c) => !c.hidden && !c.aliases.includes(c.id))
+      .sort((a, b) => a.id.localeCompare(b.id))
+
+    const commandChoices = commands.map((c) => {
+      const name = toConfiguredId(c.id, this.config)
+      return {
+        description: c.summary,
+        name,
+        value: name,
+      }
+    })
```

Comment:
since we're not using the commands variable
```suggestion
    const commandChoices = this.config.commands
      .filter((c) => !c.hidden && !c.aliases.includes(c.id))
      .sort((a, b) => a.id.localeCompare(b.id))
      .map(((c) => {
        const name = toConfiguredId(c.id, this.config)
        return {
          description: c.summary,
          name,
          value: name,
        }
    })
```